Gems/ROS2/Code/Source/Lidar/ROS2LidarSensorComponent.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "Lidar/LidarRaycaster.h"
+#include "Lidar/LidarTemplate.h"
+#include "Lidar/LidarTemplateUtils.h"
+#include "ROS2/Sensor/ROS2SensorComponent.h"
+#include <Atom/RPI.Public/AuxGeom/AuxGeomDraw.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <rclcpp/publisher.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+
+namespace ROS2
+{
+    //! Lidar sensor Component.
+    //! Lidars (Light Detection and Ranging) emit laser light and measure it after reflection.
+    //! Lidar Component allows customization of lidar type and behavior and encapsulates both simulation.
+    //! and data publishing. Lidar Component requires ROS2FrameComponent.
+    // TODO - Add selection of implementation choice (PhysX, GPU, other), noise
```

Comment:
This TODO should be made into an issue on Github: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#todo-comments